packages/common/client-utils/src/events/emitter.ts

```diff
@@ -3,68 +3,45 @@
  * Licensed under the MIT License.
  */
 
-import { UsageError } from "@fluidframework/telemetry-utils/internal";
-import { getOrCreate } from "../util/index.js";
-import type { Listenable, Listeners, Off } from "./listeners.js";
+import type {
+	Listenable,
+	Listeners,
+	Off,
+	HasListeners,
+	NoListenersCallback,
+	IEmitter,
+} from "@fluidframework/core-interfaces/internal";
 
 /**
- * Interface for an event emitter that can emit typed events to subscribed listeners.
+ * Subset of Map interface.
```

Comment:
Let's add more detail here. Specifically, this just covers the get and set methods of the Map interface.